task11.rb

```diff
@@ -0,0 +1,25 @@
+# In array of strings first character of next word must match last character of previous word and the word must not have already been said so ["word", "dowry", "yodel", "leader"] - is valid but ["some", "thing"] and ["ant", "tea", "ant"] are not
+
+def task11(arr)
+  raise ArgumentError, "Not a valid format. Need an array." if arr.class != Array
+  raise ArgumentError, "String is empty. This is not valid" if arr.empty?
+  raise ArgumentError, "Array must contain only string inside." if arr.one? {|elem| elem.class != String}
+  arr.each do |elem|
+    elem.downcase!
+    elem.strip!
+    raise ArgumentError, "The element must contain at least one letter." if arr.one? {|elem| elem.empty?}
+    raise ArgumentError, "The element must contain only one word" if !elem.match(/\A[a-zA-Z]+\z/)
+  end
+  raise ArgumentError, "Array can not contain oly one word. It is not a game thought." if arr.length == 1
+  raise ArgumentError, "Array must contain only unique values." if arr.length != arr.uniq.length
+
+  i = 0
```

Comment:
@MaraEfimova Please use an enumerator, avoid using loops when you could use enumerator(s) for the same purpose.